lib/Db/AclMapper.php

```diff
@@ -29,22 +29,53 @@
 use OCP\IDBConnection;
 
 class AclMapper extends DeckMapper implements IPermissionMapper {
+
+	/**
+	 * @param IDBConnection $db
+	 */
 	public function __construct(IDBConnection $db) {
 		parent::__construct($db, 'deck_board_acl', Acl::class);
 	}
 
+	/**
+	 * @param int $boardId
+	 * @param int|null $limit
+	 * @param int|null $offset
+	 * @return Acl[]
+	 * @throws \OCP\DB\Exception
+	 */
 	public function findAll($boardId, $limit = null, $offset = null) {
-		$sql = 'SELECT id, board_id, type, participant, permission_edit, permission_share, permission_manage FROM `*PREFIX*deck_board_acl` WHERE `board_id` = ? ';
-		return $this->findEntities($sql, [$boardId], $limit, $offset);
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('id', 'board_id', 'type', 'participant', 'permission_edit', 'permission_share', 'permission_manage')
+			->from('deck_board_acl')
+			->where($qb->expr()->eq('board_id', $qb->createNamedParameter($boardId, IQueryBuilder::PARAM_INT)))
+			->setMaxResults($limit)
+			->setFirstResult($offset);
+
+		return $this->findEntities($qb);
 	}
 
+	/**
+	 * @param int $userId
+	 * @param int $aclId
+	 * @return bool
+	 * @throws \OCP\DB\Exception
+	 */
 	public function isOwner($userId, $aclId): bool {
-		$sql = 'SELECT owner FROM `*PREFIX*deck_boards` WHERE `id` IN (SELECT board_id FROM `*PREFIX*deck_board_acl` WHERE id = ?)';
-		$stmt = $this->execute($sql, [$aclId]);
-		$row = $stmt->fetch();
-		return ($row['owner'] === $userId);
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('*')
```

Comment:
Might be very minor optimisation, but I think some databases are able to get the information just from the index when querying just an indexed column like the id of the ACL. Since we only use the rowCount that should be enough.

Reply:
Nice suggestion, added!